packages/Logger/src/index.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * MIT License
+
+ * Copyright (c) 2018 Priit Kallas
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+**/
+
+/**
+ * This interface heavily borrows from the `ts-log` package. However, the `trace`
+ * function is not on all of the common loggers that we claim to support so
+ * it's been removed in this copy of this interface.
+ */
+
+/**
+ * Represents a generic logger that could be a simple console, bunyan etc.
+ */
+export interface Logger {
+  debug(message?: any, ...optionalParams: any[]): void;
```

Comment:
I'm not sure I follow the optional params thing.  My understanding is that you should be able to write code that treats this Logger generically, ie, any code you write to it should be something you expect to work with any Logger implementation. But multiple argument handling is going to be pretty implementation-specific?

Like a lot of things in AS my philosophy is when you're writing your own app code you should use the actual implementation you're using directly; the interface is for code that needs to work with any implementation, ie mostly core code or open source plugins.

Reply:
Right now if your logger supports multiple args, TS will tell you that you can't use them (in our AS `Logger` interface) since we only declare one parameter.

Reply:
Spoke offline about this and agree these `optionalParams` don't make sense for the internal types of `Logger`. Those params will be specific per implementation and will be meaningless within the AS or Gateway context.

If users want stronger typings around the logging facilities they can use the instance of their logger directly.